File: Dog.cpp
#include <iostream>
#include "Dog.hpp"

using namespace std;

/*
Constructor yang menerima parameter position, name, meat.
Parameter diturunkan dari MeatProducing.
*/
Dog::Dog(pair<int,int> _position, string _name, Land* _landPos){
    initializeFarmAnimal(_position, _name, _landPos);
    initializeMeatProducing(new DogMeat(HARGA_DAGING_ANJING, getName(), FARMPRODUCT, DOG, 'D'));
}

/*
Method untuk mengetahui apakah Dog sudah mati
karena dagingnya diproduksi. Method diturunkan dari
MeatProducing
*/
void Dog::produceMeat(Player& _player){
    if (meat != NULL){
        _player.addInventory(meat);
        meat = NULL;
    }
}

bool isKillable(){
    return true;
}

void Dog::eat(){
    FarmAnimal::eat();
}

void Dog::talk(){
    cout << name << ": Guk Guk Ruyuk!" << endl;
}

File: Java/src/engisubs/gameobject/farmanimal/Dog.java
package engisubs.gameobject.farmanimal;

import java.util.*;

import engisubs.gameobject.*;
import engisubs.gameobject.cell.land.*;
import engisubs.gameobject.product.farmproduct.meat.*;

public class Dog extends FarmAnimal implements MeatProducing{
    public Dog(Map <String, Integer> _position, String _name, Land _landPos){
        initGameObject(GameObjectType.DOG, 'D');
        initializeFarmAnimal(_position, _name, _landPos);
    }

    public void produceMeat(Player _p){
        _p.addInventory(new DogMeat(50, name));
    }

    public void talk(){
        System.out.println(name + ": Woof Rawr XD");
    }

    public void interact(Player _p){
        System.out.println("You cannot do anything with a Dog, or can you? *wink*");
    }
}